pySDC/implementations/sweeper_classes/generic_implicit_MPI.py

```diff
@@ -127,12 +127,20 @@ def predict(self):
         # evaluate RHS at left point
         L.f[0] = P.eval_f(L.u[0], L.time)
 
+        m = self.rank
+
         if self.params.initial_guess == 'spread':
-            L.u[self.rank + 1] = P.dtype_u(L.u[0])
-            L.f[self.rank + 1] = P.eval_f(L.u[self.rank + 1], L.time + L.dt * self.coll.nodes[self.rank])
+            # copy u[0] to all collocation nodes, evaluate RHS
+            L.u[m + 1] = P.dtype_u(L.u[0])
+            L.f[m + 1] = P.eval_f(L.u[m + 1], L.time + L.dt * self.coll.nodes[m])
+        elif self.params.initial_guess == 'copy':
+            # copy u[0] and RHS evaluation to all collocation nodes
+            L.u[m] = P.dtype_u(L.u[0])
```

Comment:
Why is it `m+1` in the other cases but `m` here? Did you test that this works? I would guess not.

Reply:
Just took the previous implementation (which was not tested, btw ...). Due to this ghost point in the Q matrix, that is not in the nodes ....

Reply:
But there was no previous implementation of your new feature :D I am 72% sure that this will fail. Please test it!